lib/kubernetes-deploy/deploy_task.rb

```diff
@@ -557,6 +557,27 @@ def confirm_namespace_exists
       @logger.info("Namespace #{@namespace} found")
     end
 
+    # make sure to never prune the ejson-keys secret
+    def confirm_ejson_keys_not_prunable
+      out, err, st = kubectl.run("get", "secret", EjsonSecretProvisioner::EJSON_KEYS_SECRET,
```

Comment:
Even after this refactor, we're still making this request (and parsing the resulting data) twice in the case where they do have ejson secrets--once here and once when the provisioner needs it. Can we make the provisioner expose it instead? That's what I thought we agreed to last week (though I forgot about it on my last review).